cmd2/cmd2.py

```diff
@@ -3332,18 +3332,17 @@ def do_history(self, args: argparse.Namespace) -> None:
             for hi in history:
                 self.poutput(hi.pr(script=args.script, expanded=args.expanded, verbose=args.verbose))
 
-    def _generate_transcript(self, history: List[HistoryItem], transcript_file: str) -> None:
+    def _generate_transcript(self, history: List[Union[HistoryItem, str]], transcript_file: str) -> None:
```

Comment:
It doesn't look like `transcript_file` is ever run through `os.path.expanduser()`.
Sort of related, I'm also not sure if the paths in `self.transcript_files` are expanded either.

Reply:
Good call!